packages/e2e/test/k6/scenarios/web-socket.test.js

```diff
@@ -92,23 +105,31 @@ export const run = ({ wallets }) => {
         return socket.setTimeout(() => {
           closed = true;
           socket.close();
-        }, 1000);
+        }, idleSeconds * 1000);
       }
 
       const address =
         requestId <= wallet.length
           ? wallet[requestId - 1].address
           : getDummyAddr(wallet[0].address, requestId - wallet.length);
 
-      socket.send(JSON.stringify({ requestId, txsByAddresses: { addresses: [address], lower: 0 } }));
+      const lower = walletRestoration ? 0 : blockNo;
+
+      socket.send(JSON.stringify({ requestId, txsByAddresses: { addresses: [address], lower } }));
     };
 
     socket.on('message', (message) => {
-      const { clientId, responseTo, transactions } = JSON.parse(message);
+      const { clientId, networkInfo, responseTo, transactions } = JSON.parse(message);
 
+      // Set operational stat
       if (clientId) operationalTrend.add(Date.now() - begin);
+
+      // Perform init with or without restoration
+      if (networkInfo) ({ blockNo } = networkInfo.ledgerTip);
```

Comment:
What is this syntax? I don't think I've ever seen it before. If I had to guess, you are deconstructing `ledgerTip` and setting variable `blockNo` defined on line 93?

Reply:
Yes